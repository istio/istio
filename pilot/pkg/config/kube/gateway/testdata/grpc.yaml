apiVersion: gateway.networking.k8s.io/v1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
      kinds:
      - kind: GRPCRoute
---
apiVersion: gateway.networking.k8s.io/v1
kind: GRPCRoute
metadata:
  name: grpc
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["first.domain.example", "another.domain.example"]
  rules:
  - matches:
    - method:
        service: "foo"
      headers:
      - name: my-header
        value: some-value
        type: Exact
    filters:
    - type: RequestHeaderModifier
      requestHeaderModifier:
        add:
        - name: my-added-header
          value: added-value
        remove: [my-removed-header]
    backendRefs:
    - name: httpbin
      port: 80
  - matches:
    - method:
        type: RegularExpression
        method: "bar"
    backendRefs:
    - name: httpbin
      port: 80
