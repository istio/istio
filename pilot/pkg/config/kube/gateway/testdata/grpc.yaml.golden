apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: GRPCRoute/grpc.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~another.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - another.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /foo/
    name: default.grpc.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        regex: /[^/]+/bar
    name: default.grpc.1
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: GRPCRoute/grpc.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: istio-system~gateway-istio-autogenerated-k8s-gateway-default~first.domain.example
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - first.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /foo/
    name: default.grpc.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        regex: /[^/]+/bar
    name: default.grpc.1
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
